packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -320,90 +253,43 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
 				};
 			}
 		}
+
+		if (childChange !== undefined) {
+			inverted.childChange = childChange;
+		}
+
 		return inverted;
 	},
 
 	rebase: (
 		change: OptionalChangeset,
 		overChange: OptionalChangeset,
 		rebaseChild: NodeChangeRebaser,
+		_genId: IdAllocator,
+		nodeManager: RebaseNodeManager,
 	): OptionalChangeset => {
-		const { moves, childChanges, valueReplace: field } = change;
-
-		// TODO: avoid computing the dstToSrc map if it's not needed.
-		// TODO: de-dupe overSrcToDst and forwardMap
-		const { srcToDst: overSrcToDst } = getBidirectionalMaps(overChange.moves);
-
-		const forwardMap = new RegisterMap<RegisterId>();
-		for (const [src, dst] of overChange.moves) {
-			forwardMap.set(src, dst);
-		}
-		if (overChange.valueReplace !== undefined) {
-			const effectfulDst = getEffectfulDst(overChange.valueReplace);
-			if (effectfulDst !== undefined) {
-				forwardMap.set("self", overChange.valueReplace.dst);
-			}
-			if (overChange.valueReplace.src !== undefined) {
-				forwardMap.set(overChange.valueReplace.src, "self");
-			}
-		}
-
-		const rebasedMoves: Move[] = [];
-		for (const [src, dst] of moves) {
-			const newDst = tryGetFromNestedMap(overSrcToDst, src.revision, src.localId);
-			rebasedMoves.push([src, newDst ?? dst]);
-		}
+		const rebased: Mutable<OptionalChangeset> = {};
 
-		const overChildChangesBySrc = new RegisterMap<NodeId>();
-		for (const [id, childChange] of overChange.childChanges) {
-			overChildChangesBySrc.set(id, childChange);
+		const baseChild = getBaseChildChange(overChange, nodeManager);
+		const rebasedChild = rebaseChild(change.childChange, baseChild);
+		if (isReplaceEffectful(overChange.valueReplace) && !overChange.valueReplace.isEmpty) {
+			// The child node was detached by overChange.
+			nodeManager.rebaseOverDetach(overChange.valueReplace.dst, 1, undefined, rebasedChild);
```

Comment:
I'm curious if we sometimes should transfer the detach that's here to the MCF. I'm thinking about scenarios where the detach should follow the node because we want to attach the node somewhere else. This can happen with undo right now I think.